aldryn_people/cms_toolbar.py

```diff
@@ -7,10 +7,38 @@
 from cms.toolbar_base import CMSToolbar
 from cms.toolbar_pool import toolbar_pool
 from cms.utils.urlutils import admin_reverse
+from parler.models import TranslatableModel
 
 from .models import Group, Person
 
 
+def get_obj_from_request(model, request,
+                         pk_url_kwarg='pk',
+                         slug_url_kwarg='slug'):
+    """
+    Given a model and the request, try to extract and return an object
+    from an available 'pk' or 'slug', or return None.
+
+    Note that no checking is done that the view's kwargs really are for objects
+    matching the provided model (how would it?) so use only where appropriate.
+    """
+    language = get_language_from_request(request, check_path=True)
+    kwargs = request.resolver_match.kwargs
+    qs = model.objects
+    if pk_url_kwarg in kwargs:
+        return qs.filter(pk=kwargs[pk_url_kwarg]).first()
+    elif slug_url_kwarg in kwargs:
+        # If the model is translatable, and the given slug is a translated
+        # field, then find it the Parler way.
+        if (issubclass(model, TranslatableModel) and
+                slug_url_kwarg in model._parler_meta.get_translated_fields()):
+            return qs.translated(language, slug=kwargs[slug_url_kwarg]).first()
```

Comment:
you're hardcoding ```slug``` here which means that if ```slug_url_kwarg``` points to a name that's not ```slug```, field error will be raised.